src/client.c

```diff
@@ -31,23 +31,30 @@ urlinfo_t *parse_url(char *url)
   char *hostname = strdup(url);
   char *port;
   char *path;
+  char *backlash, *colon;
 
   urlinfo_t *urlinfo = malloc(sizeof(urlinfo_t));
-
-  /*
-    We can parse the input URL by doing the following:
-
-    1. Use strchr to find the first backslash in the URL (this is assuming there is no http:// or https:// in the URL).
-    2. Set the path pointer to 1 character after the spot returned by strchr.
-    3. Overwrite the backslash with a '\0' so that we are no longer considering anything after the backslash.
-    4. Use strchr to find the first colon in the URL.
-    5. Set the port pointer to 1 character after the spot returned by strchr.
-    6. Overwrite the colon with a '\0' so that we are just left with the hostname.
-  */
-
-  ///////////////////
-  // IMPLEMENT ME! //
-  ///////////////////
+  
+  // We can parse the input URL by doing the following:
+  
+  // 1. Use strchr to find the first backslash in the URL (this is assuming there is no http:// or https:// in the URL).
+  backlash = strchr(hostname, '/');
```

Comment:
This is where I probably would've used strstr() instead of strchr(). To make sure you capture both instances of a forwardslash in a url '://' you can actually target the colon and two forward slashes all in one step